skrub/_table_vectorizer.py

```diff
@@ -32,79 +35,34 @@
-        frac=min(n_trials / len(date_column), 1), random_state=42
-    )
-    # try to infer the date format
-    # see if either dayfirst or monthfirst works for all the rows
-    with warnings.catch_warnings():
-        # pandas warns when dayfirst is not strictly applied
-        warnings.simplefilter("ignore")
-        date_format_monthfirst = date_column_sample.apply(
-            lambda x: guess_datetime_format(x)
-        )
-        date_format_dayfirst = date_column_sample.apply(
-            lambda x: guess_datetime_format(x, dayfirst=True),
-        )
-    # if one row could not be parsed, return None
-    if date_format_monthfirst.isnull().any() or date_format_dayfirst.isnull().any():
-        return
-    # even with dayfirst=True, monthfirst format can be inferred
-    # so we need to check if the format is the same for all the rows
-    elif date_format_monthfirst.nunique() == 1:
-        # one monthfirst format works for all the rows
-        # check if another format works for all the rows
-        # if so, raise a warning
-        if date_format_dayfirst.nunique() == 1:
-            # check if monthfirst and dayfirst haven't found the same format
-            if date_format_monthfirst.iloc[0] != date_format_dayfirst.iloc[0]:
-                warnings.warn(
-                    f"""
-                    Both {date_format_monthfirst.iloc[0]} and
-                    {date_format_dayfirst.iloc[0]} are valid formats for the dates in
-                    column '{date_column.name}'.
-                    Format {date_format_monthfirst.iloc[0]} will be used.
-                    """,
-                    UserWarning,
-                    stacklevel=2,
-                )
-        return date_format_monthfirst.iloc[0]
-    elif date_format_dayfirst.nunique() == 1:
-        # only this format works for all the rows
-        return date_format_dayfirst.iloc[0]
-    else:
-        # more than two different formats were found
-        # TODO: maybe we could deal with this case
-        return
-
-
-def _has_missing_values(df: pd.DataFrame | pd.Series) -> bool:
-    """
-    Returns True if `array` contains missing values, False otherwise.
+    X : pandas.DataFrame
     """
-    return any(df.isnull())
+    X_out = dict()
+    for col in X.columns:
+        if not is_datetime64_any_dtype(X[col]):
+            # We don't use errors="ignore" because it casts string
+            # and categories to object dtype in Pandas < 2.0.
+            # TODO: replace 'raise' by 'ignore' and remove the exception
+            # catching when the minimum pandas version of skrub is 2.0.
+            try:
+                X_out[col] = pd.to_numeric(X[col], errors="raise")
+                continue
+            except (ValueError, TypeError):
+                pass
+        X_out[col] = X[col]
+    return pd.DataFrame(X_out, index=X.index)
 
 
-def _replace_false_missing(df: pd.DataFrame | pd.Series) -> pd.DataFrame | pd.Series:
+def _replace_false_missing(df):
```

Comment:
in practice it seems to be only used with series as input. can we change the docstring to say it only takes series? I  think it would reduce a bit the cognitive load when reading it and it will ease porting to polars or the dataframe api too